pandas/tseries/tools.py

```diff
@@ -510,6 +510,9 @@ def coerce(values):
         # we allow coercion to if errors allows
         return to_numeric(values, errors=errors)
 
+    # prevent overflow in case of int8 or int16
+    arg = arg.astype('int64', copy=False)
```

Comment:
good point. @ravinimmi can you add checking that the passes dtypes are all integer.

use ``is_integer_dtype``